### PR TITLE
Update ledger highlight queries

### PR DESCRIPTION
**Pull request by cbarrete**
This is a reboot of https://github.com/nvim-treesitter/nvim-treesitter/pull/4328, which apparently stalled due to some setup issues on @afh's side.
Highlights are currently slightly broken for ledger, so unless the other PR sees new activity, I think that this should be merged.

### CODE REVIEW

**Comment by amaanq** [comment]
the fine-tuning of using sub captures everywhere to distinguish every little node isn't something we would like to see here, that's more a user's custom configuration

please keep it simple and group things accordingly, with captures not having a subcapture not supported in our CONTRIBUTING.MD

**Comment by cbarrete** [comment]
Thanks for the feedback @amaanq, does it look acceptable now?

Also tagging @afh, who might want to still have the original changes in their config.

File: queries/ledger/highlights.scm [hunk 1]
```diff
@@ -1,16 +1,26 @@
...
+    (block_comment)
     (comment)
     (note)
+    (test)
 ] @comment
 
 [
     (date)
+    (effective_date)
+    (time)
```
Comment (ObserverOfTime) [inline queries/ledger/highlights.scm:11]: Are these numeric? If not, they should be `@constant`.

Comment (cbarrete) [reply queries/ledger/highlights.scm:11]: Yes they are, e.g. `2023-04-23`. They especially feel numeric with something like [tpope/vim-speeddating](https://github.com/tpope/vim-speeddating)

Comment (ObserverOfTime) [reply queries/ledger/highlights.scm:11]: `2023-04-23` is not entirely numeric.

Comment (cbarrete) [reply queries/ledger/highlights.scm:11]: This is in the context of ledger, which is not a programming language but a language for describing financial transactions.
In this context, it *feels* much more like a numeral than a constant, string, etc.

Comment (cbarrete) [reply queries/ledger/highlights.scm:11]: I guess @string.special could work here too given the description in CONTRIBUTING.md

Comment (afh) [reply queries/ledger/highlights.scm]: Given that dates generally may include month and day names `@string.special` seems very fitting, especially as [`CONTRIBUTING.md` §Literals](https://github.com/nvim-treesitter/nvim-treesitter/blob/master/CONTRIBUTING.md#literals) calls dates out explicitly

**Comment by afh** [comment]
Thanks for jumping in on this @cbarrete, I'd much prefer help with my setup and contributions over on #4328 than see this reboot. That said, if this reboot is what it takes to get the improvements in I'll be okay with it.

@amaanq I understand the dislike of _"sub captures everywhere to distinguish every little node"_ as a ledger nvim tree-sitter user I'd like much flexibility with highlighting. Where could I find examples of user custom configurations that allow for fine-grained control over highlighting with fewer sub captures?

**Comment by cbarrete** [comment]
> Thanks for jumping in on this @cbarrete, I'd much prefer help with my setup and contributions over on #4328 than see this reboot. That said, if this reboot is what it takes to get the improvements in I'll be okay with it.

That's fine with me as long as it can be resolved quickly enough.

**Comment by afh** [comment]
The changes here seem to be missing the highlights for ledger's timelogging, i.e. `check_in` and `check_out`. Was this intentional? If yes, what was the motivation for the removal @cbarrete? 🙂

**Comment by cbarrete** [comment]
Just a mistake on my side :)

**Review by ObserverOfTime — APPROVED** [review]

**Comment by afh** [comment]
Gotcha, thanks for addressing this so quickly, much appreciated :+1:

**Comment by amaanq** [comment]
> @amaanq I understand the dislike of _"sub captures everywhere to distinguish every little node"_ as a ledger nvim tree-sitter user I'd like much flexibility with highlighting. Where could I find examples of user custom configurations that allow for fine-grained control over highlighting with fewer sub captures?

You can write your own query files and do whatever you like in your own config, no one is stopping you there :wink: it just does not belong here.

**Comment by afh** [comment]
@amaanq I may have misunderstood this project's intent then, as I thought it was meant to allow users to configure the highlighting to their own desires with fine granularity.

For my own understanding what are the disadvantages of the sub captures o distinguish every little node?

**Comment by amaanq** [comment]
tree-sitter queries allow you to configure it however you desire, but nvim-treesitter is a central repo to generate generic captures that can be applied to most if not all languages, keeping a non-haphazard standard makes it easy for themes to support out of the box.

**Comment by afh** [comment]
I see, thanks for providing the helpful context, @amaanq.

My focus currently lies mainly with [ledger](https://github.com/ledger/ledger). Once that project is has seen a few releases with the work I intend to contribute and I look for other interesting projects I'll possibly come back to tree-sitter and the matrix chat 🙂 Keep up the good work!!

**Pull request merged**